skfda/exploratory/visualization/representation.py

```diff
@@ -305,23 +318,27 @@ def plot(
             for i in range(self.fdata.dim_codomain):
                 for j in range(self.fdata.n_samples):
 
-                    set_color_dict(sample_colors, j, color_dict)
+                    set_color_dict(self.sample_colors, j, color_dict)
```

Comment:
Do not modify attributes in `plot`. Create a new dict if necessary.

Reply:
no update in the code here, i just moved it into a function as it was giving me warning in flake8 for complexity